notebooks/argo-file-checker.ipynb

```diff
@@ -0,0 +1,417 @@
+    "HEALTH_ENDPOINT = \"/\"\n",
+    "\n",
+    "# Request settings\n",
+    "TIMEOUT = 30  # API request timeout in seconds\n",
+    "HEADERS = {\n",
+    "    \"accept\": \"application/json\"\n",
+    "}\n",
+    "\n",
+    "# Full URL for file check endpoint\n",
+    "FILE_CHECK_URL = f\"{API_BASE_URL}/{CHECK_FILE_ENDPOINT}\"\n",
+    "\n",
+    "logger.info(\"🔗 API Base URL:%s\",  API_BASE_URL)\n",
+    "logger.info(\"🏛️ DAC: %s\" ,DEFAULT_DAC)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 102,
+   "id": "48a5fa1a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ============================================\n",
+    "#  3️⃣ function to test config and api\n",
+    "# ============================================\n",
+    "def test_api_connection() -> bool:\n",
+    "    \"\"\"Test if the API is accessible.\"\"\"\n",
+    "    logger.info(\"\\n🔍 Testing API Connection..\")\n",
+    "    logger.info(\"-\" * 30)\n",
+    "\n",
+    "    try:\n",
+    "        response = requests.get(f\"{API_BASE_URL}/\", timeout=5)\n",
+    "        if response.status_code != 200:\n",
+    "            logger.info(\"API returned status code: %s\",response.status_code)\n",
+    "            return False\n",
+    "        else:\n",
+    "            result = response.json()\n",
+    "            logger.info(\"API is accessible!\")\n",
+    "            logger.info(\"Health Check Response: %s\",result)\n",
+    "            return True\n",
+    "\n",
+    "\n",
+    "\n",
+    "    except requests.exceptions.ConnectionError as error:\n",
+    "        logger.info(\"Could not connect to API. Is the container running?\")\n",
+    "        logger.info(error)\n",
+    "        return False\n",
+    "\n",
+    "    except requests.exceptions.Timeout:\n",
+    "        logger.info(\"Connection timed out\")\n",
+    "        return False\n",
+    "\n",
+    "    except Exception as e:\n",
+    "        logger.exception(\"Unexpected error\")\n",
+    "        return False\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 97,
+   "id": "28b3891c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ============================================\n",
+    "#  4️⃣ function to check the files\n",
+    "# ============================================\n",
+    "def file_check(file_paths:list[str], dac:str=DEFAULT_DAC)-> dict[str, Any]:\n",
+    "    \"\"\"Filechecker to check nc files.\"\"\"\n",
+    "    files_data = []\n",
+    "\n",
+    "    for file_path in file_paths:\n",
+    "        path = Path(file_path)\n",
+    "        if not path.exists():\n",
+    "            logger.info(\"File not found: %s\",file_path)\n",
+    "            continue\n",
+    "\n",
+    "        # Determine content type based on file extension\n",
+    "        content_type = {\n",
+    "            \".json\": \"application/json\",\n",
+    "            \".nc\": \"application/octet-stream\",\n",
```

Comment:
`application/x-netcdf` is the convention for NetCDF data, I believe. Does the API recognise that?